tools/config.py

```diff
@@ -797,6 +798,13 @@ def validate_config(self):
         return True
 
 
+    @property
+    def name(self):
+        if "artifact_name" in self.app_config_data:
```

Comment:
I just thought of that based on seeing related issues in the past  (ARMmbed/mbed-cli#229). Not sure if `mbed_lib.json` is used with `mbed compile --library`? 
Also not sure whether working for both is needed for consistency?